lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -1464,16 +1459,12 @@ def parse_severity(severity_str)
 
       # If the severity is an integer (string) return it as an integer,
       # truncated to the closest valid value (multiples of 100 between 0-800).
-      if /\A\d+\z/.match(severity)
+      if /\A\d+\z/ =~ severity
         begin
           numeric_severity = (severity.to_i / 100) * 100
-          if numeric_severity < 0
-            return 0
-          elsif numeric_severity > 800
-            return 800
-          else
-            return numeric_severity
-          end
+          return 0 if numeric_severity < 0
```

Comment:
[Optional] Would this be more readable?
```ruby
case
when numeric_severity < 0
  return 0
when numeric_severity > 800
  return 800
else
  return numeric_severity
end
```
?

Reply:
Weird -- it was ok with that for me:
```diff
$ git diff origin/lingshi-update-rubocop 
diff --git a/lib/fluent/plugin/out_google_cloud.rb b/lib/fluent/plugin/out_google_cloud.rb
index b78eb42..a22e3ea 100644
--- a/lib/fluent/plugin/out_google_cloud.rb
+++ b/lib/fluent/plugin/out_google_cloud.rb
@@ -1421,8 +1421,9 @@ module Fluent
     end
 
     # Values permitted by the API for 'severity' (which is an enum).
-    VALID_SEVERITIES = Set.new(%w(DEFAULT DEBUG INFO NOTICE WARNING ERROR
-                                  CRITICAL ALERT EMERGENCY)).freeze
+    VALID_SEVERITIES = Set.new(
+      %w(DEFAULT DEBUG INFO NOTICE WARNING ERROR CRITICAL ALERT EMERGENCY)
+    ).freeze
 
     # Translates other severity strings to one of the valid values above.
     SEVERITY_TRANSLATIONS = {
@@ -1465,9 +1466,14 @@ module Fluent
       if /\A\d+\z/ =~ severity
         begin
           numeric_severity = (severity.to_i / 100) * 100
-          return 0 if numeric_severity < 0
-          return 800 if numeric_severity > 800
-          return numeric_severity
+          case
+          when numeric_severity < 0
+            return 0
+          when numeric_severity > 800
+            return 800
+          else
+            return numeric_severity
+          end
         rescue
           return 'DEFAULT'
         end
```
```
$ rubocop _0.39.0_ lib/fluent/plugin/out_google_cloud.rb 
Inspecting 1 file
.

1 file inspected, no offenses detected
```
What error did you get?

Reply:
Ah, I was probably getting the error in a different `rubocop` version. It was this one that I got:
```
'Use a guard clause instead of wrapping the code inside a conditional expression.'
```
from http://www.rubydoc.info/github/bbatsov/rubocop/Rubocop/Cop/Style/GuardClause.

I just reverted the change since `v0.39.0` does not have this requriement.

Reply:
I think `v0.39.0` wasn't happy with the original `if`/`elseif`/`else`, but was ok with `case`/`when`.